sale_advance_payment/models/payment.py

```diff
@@ -0,0 +1,12 @@
+# Copyright 2017 Omar Castiñeira, Comunitea Servicios Tecnológicos S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountPayment(models.Model):
+    _inherit = "account.payment"
+
+    sale_id = fields.Many2one(
+        "sale.order", "Sale", readonly=True, states={"draft": [("readonly", False)]}
+    )
```

Comment:
Since Odoo 17, property `states` is no longer supported.
You should define the `readonly` attribute on the related views instead.
something of the like of `<field name="sale_id" readonly="state != 'draft'">`